### PR TITLE
chore: upgrade some dependencies to the latest version

### PR DESCRIPTION
**Pull request by joos2010kj**
List of yarn-upgraded dependencies:

@types/classnames
@types/react-css-modules
gh-pages
papaparse
react@^15.5.4
redux-undo@1.0.0-beta9-9-7
tslib
vega-datasets@1.8 -> ^1.31.0
webpack@2.5.1 -> ^2.7.0

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -131,8 +141,8 @@
     "tslint-react": "^2.3.0",
     "typescript": "^2.3.4",
     "url-loader": "^0.5.8",
-    "webpack": "^2.5.1",
-    "webpack-dev-server": "^2.11.2",
+    "webpack": "2.7.0",
```
Comment (domoritz) [inline package.json:144]:
```suggestion
    "webpack": "^2.7.0",
```

File: package.json [hunk 2]
```diff
@@ -65,21 +72,24 @@
...
     "vega": "4.0.0",
-    "vega-datasets": "^1.8.0",
+    "vega-datasets": "^2.2.0",
     "vega-expression": "^2.3.1",
     "vega-lite": "2.4.0",
     "vega-tooltip": "^0.4.2",
     "vega-typings": "^0.2.15",
-    "vega-util": "1.7.0"
+    "vega-util": "1.7.0",
+    "webpack-sources": "1.0.1"
```
Comment (domoritz) [inline package.json:89]: Is this really a dep?

File: package.json [hunk 3]
```diff
@@ -65,21 +72,24 @@
...
     "redux-thunk": "^2.2.0",
-    "redux-undo": "1.0.0-beta9-9-6",
+    "redux-undo": "1.0.0-beta9-9-7",
     "reselect": "^2.5.4",
+    "source-list-map": "2.0.0",
     "tether": "^2.0.0",
     "throttle-debounce": "^3.0.1",
-    "tslib": "^2.2.0",
+    "tslib": "^2.3.0",
+    "uglify-js": "2.8.27",
```
Comment (domoritz) [inline package.json:81]: This is not a dep

File: package.json [hunk 4]
```diff
@@ -37,15 +37,22 @@
...
+    "@types/fast-json-stable-stringify": "2.0.0",
+    "acorn": "5.0.0",
     "ajv": "^5.2.2",
+    "clone": "2.1.2",
     "compassql": "0.20.2",
+    "enhanced-resolve": "3.3.0",
+    "fast-deep-equal": "2.0.1",
     "font-awesome": "^4.7.0",
     "isomorphic-fetch": "^3.0.0",
+    "json-stringify-pretty-compact": "2.0.0",
```
Comment (domoritz) [inline package.json:50]: Why do we need these deps?

**Review by domoritz — CHANGES_REQUESTED** [review]

File: .vscode/tasks.json [hunk 5]
```diff
@@ -1,20 +1,31 @@
 {
-  "version": "0.1.0",
+  "version": "2.0.0",
```
Comment (domoritz) [inline .vscode/tasks.json:2]: Please don't bump the version

Comment (joos2010kj) [reply .vscode/tasks.json:2]: Sorry, I'm not sure why it got bumped b/c I didn't touch it...

Comment (domoritz) [reply .vscode/tasks.json:2]: Oh, nm. This is actually okay. I somehow thoughts this was the package.json.

**Comment by joos2010kj** [comment]
Let me push again with the same changes but using WSL b/c this previous one was done over Windows due to a Git error I had that I just fixed.

File: package.json [hunk 6]
```diff
@@ -42,10 +42,10 @@
     "font-awesome": "^4.7.0",
     "isomorphic-fetch": "^3.0.0",
     "moment": "^2.29.1",
-    "papaparse": "^5.3.0",
+    "papaparse": "^5.3.1",
     "rc-slider": "^8.2.0",
     "re-reselect": "^4.0.0",
-    "react": "^15.4.2",
+    "react": "15.5.4",
```
Comment (domoritz) [inline package.json:48]:
```suggestion
    "react": "^15.5.4",
```

File: package.json [hunk 7]
```diff
@@ -65,21 +65,22 @@
     "redux-action-log": "^2.0.1",
     "redux-logger": "^3.0.6",
     "redux-thunk": "^2.2.0",
-    "redux-undo": "1.0.0-beta9-9-6",
+    "redux-undo": "1.0.0-beta9-9-7",
     "reselect": "^2.5.4",
+    "source-list-map": "2.0.0",
```
Comment (domoritz) [inline package.json:70]: This should be a dev dep

File: package.json [hunk 8]
```diff
@@ -65,21 +65,22 @@
...
     "tether": "^2.0.0",
     "throttle-debounce": "^3.0.1",
-    "tslib": "^2.2.0",
+    "tslib": "^2.3.0",
     "vega": "4.0.0",
-    "vega-datasets": "^1.8.0",
+    "vega-datasets": "^2.2.0",
```
Comment (domoritz) [inline package.json:75]: Did you try voyager? I think a lot of fields have changed names here.

Comment (joos2010kj) [reply package.json:75]: I noticed that some of new vega dataset are csv files, but currently it is set to receive only json files (response => response.json()).  How should I add a csv reader?

Comment (domoritz) [reply package.json:75]: I'd stay with Vega datasets v1 for now.

Comment (joos2010kj) [reply package.json:75]: OK.  

In version 1.9 through 1.31 (latest of v1), these are introduced:
disasters.csv
co2-concentration.csv
weather.csv
us-unemployment.csv
windvectors.csv

earthquakes.json
us-state-capitals.json
uniform-2d.json
volcano.json
annual-precip.json
ohlc.json

Would you like me to:
1) Keep it at the original version, 1.8
2) Upgrade to the latest of v1.x, but make only json datasets accessible.

Comment (domoritz) [reply package.json:75]: Latest1.x is fine

Comment (joos2010kj) [reply package.json]: Upgraded to 1.31.0.

Only the following three were added because the others weren't in an array format, which is a data type requirement:

- US-state-capitals
- Uniform-2d
- OHLC


image

**Comment by joos2010kj** [comment]
Please let me know if you see any final changes I need to make.

File: .vscode/tasks.json.old [hunk 9]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (domoritz) [inline .vscode/tasks.json.old:1]: remove this

File: src/constants.ts [hunk 10]
```diff
@@ -81,6 +81,21 @@ export const DEFAULT_DATASETS = [
   url: 'data/weball26.json',
   id: 'weball26',
   group: 'sample'
+}, {
+  name: 'US-state-capitals',
+  url: 'data/us-state-capitals.json',
+  id: 'us-state-capitals',
+  group: 'sample'
+}, {
+  name: 'Uniform-2d',
```
Comment (domoritz) [inline src/constants.ts:90]: I don't think this is a useful example

**Review by domoritz — CHANGES_REQUESTED** [review]

File: src/constants.ts [hunk 11]
```diff
@@ -81,6 +81,16 @@ export const DEFAULT_DATASETS = [
   url: 'data/weball26.json',
   id: 'weball26',
   group: 'sample'
+}, {
+  name: 'US-state-capitals',
```
Comment (domoritz) [inline src/constants.ts:85]: Can you show a screenshot of what these look like in voyager

**Review by domoritz — APPROVED** [review]

File: src/constants.ts (hunk 11 above)
Comment (joos2010kj) [reply src/constants.ts:85]: US:
image

image

OHLC:
image


Main:
image

Comment (domoritz) [reply src/constants.ts:85]: They are not useful I think. Please remove them.

Comment (joos2010kj) [reply src/constants.ts:85]: By them, do you mean just US or both US and OHLC data?

Comment (joos2010kj) [reply src/constants.ts:85]: (poke in case you missed it)

File: src/constants.ts [hunk 12]
```diff
@@ -81,6 +81,16 @@ export const DEFAULT_DATASETS = [
...
+  url: 'data/us-state-capitals.json',
+  id: 'us-state-capitals',
+  group: 'sample'
+}, {
+  name: 'OHLC',
```
Comment (domoritz) [inline src/constants.ts:90]: please remove these changes as they are unrelated to package updates

**Comment by domoritz** [comment]
Right now, this pull request only does minor and patch version updates so it should be pretty save to merge once the new examples are removed.
